src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1709,6 +1724,23 @@ AssignTasksToConnectionsOrWorkerPool(DistributedExecution *execution)
 												sizeof(TaskPlacementExecution *));
 		shardCommandExecution->placementExecutionCount = placementExecutionCount;
 
+		TupleDestination *tupleDest = task->tupleDest ?
```

Comment:
This function is already lengthy, it would have been nice to use a function for this new added part